### DISCUSSION TITLE
Mobile Particle

### DISCUSSION THREAD

**Discussion by mnabilbhuiyan** [idaholab/moose#17095] [Q&A Modules: Phase field] [answered]
Hello MOOSE users,
Accept my warm greetings. I am trying to simulate grain coarsening with mobile particles. I am following this slide. I badly need some guidance how to make these particles mobile. We want to see during coarsening if these particles change their positions how the micro structure will be changed.
![image](https://user-images.githubusercontent.com/60020086/108767037-bd3dd180-750a-11eb-8826-afedf814965d.png).
my kernel as following:
**[Kernels]
  **# Kernel block, where the kernels defining the residual equations are set up.**
  [./PolycrystalKernel]
   c = c
    # Custom action creating all necessary kernels for grain growth.  All input parameters are up in GlobalParams
  [../]
[]
[AuxVariables]
  [./c]
    [./InitialCondition]
    type = MultiSmoothCircleIC
    variable = c
    invalue = 1.0
    outvalue = 0.0001
    bubspac = 30.0 # This spacing is from bubble center to bubble center
    numbub = 25
    radius = 20.0
    int_width = 1.0
    radius_variation = 0.2
    radius_variation_type = uniform
    [../]
    [../]**

Best Regards,
Nabil

**Comment by SudiptaBiswas** [comment, marked as answer]
First, you might consider using `PolycrystalVoronoiVoidIC` to set up the initial conditions. Next, you need to include the Cahn-Hilliard equation in the system to capture the evolution of the particles/bubbles. The mobility corresponding to the CH equations would dictate the movement of these particles.

**Reply by mnabilbhuiyan** [reply]
Thank you
